x-pack/plugins/maps/common/content_management/v1/types.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type {
+  GetIn,
+  GetResult,
+  CreateIn,
+  CreateResult,
+  SearchIn,
+  SearchResult,
+  UpdateIn,
+  UpdateResult,
+  DeleteIn,
+  DeleteResult,
+} from '@kbn/content-management-plugin/common';
+import { MapContentType } from '../types';
+
+interface Reference {
+  type: string;
+  id: string;
+  name: string;
+}
+
+/* eslint-disable-next-line @typescript-eslint/consistent-type-definitions */
+export type MapSavedObjectAttributes = {
+  title: string;
+  description?: string;
+  mapStateJSON?: string;
+  layerListJSON?: string;
+  uiStateJSON?: string;
+};
+
+export interface MapItem {
+  id: string;
+  type: string;
+  version?: string;
+  createdAt?: string;
+  updatedAt?: string;
+  error?: {
```

Comment:
Seems odd to pass the error as part of the MapItem. Should this be considered tech debt?

Reply:
I added the fields from SavedObject that made sense and could be used in the future without having to create a new version of the content. `error` is part of saved object documents returned so I added it. Not sure it hurts having it there?